app/controllers/api/v2/trends_controller.rb

```diff
@@ -0,0 +1,44 @@
+module Api
+  module V2
+    class TrendsController < V2::BaseController
+      include Foreman::Controller::Parameters::Trend
+
+      before_action :find_resource, :only => [:show, :destroy]
+
+      api :GET, "/trends/", N_("List of trends counters")
+      def index
+        @trends = resource_scope_for_index
+      end
+
+      api :GET, "/trends/:id/", N_("Show a trend")
+      param :trend_id, :identifier, :required => true
+      def show
+      end
+
+      api :POST, "/trends/", N_("Create a trend counter")
+      def create
+        @trend = Trend.build_trend(trend_params)
```

Comment:
`trend_params` should be more strict about what values we require, when I don't send trendable type, I receive a following trace

```
2018-01-16T08:26:58 962c054f [app] [W] Action failed
 | NoMethodError: undefined method `constantize' for nil:NilClass
 | /home/ares/Projekty/Zdrojaky/foreman/app/models/trends/foreman_trend.rb:17:in `create_values'
 | /home/ares/Projekty/Zdrojaky/foreman/app/models/trends/foreman_trend.rb:13:in `create_values'
 | /home/ares/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/activesupport-5.1.4/lib/active_support/callbacks.rb:413:in `block in make_lambda'
 | /home/ares/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/activesupport-5.1.4/lib/active_support/callbacks.rb:235:in `block in halting_and_conditional'
 | /home/ares/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/activesupport-5.1.4/lib/active_support/callbacks.rb:511:in `block in invoke_after'
 | /home/ares/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/activesupport-5.1.4/lib/active_support/callbacks.rb:511:in `each'
 | /home/ares/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/activesupport-5.1.4/lib/active_support/callbacks.rb:511:in `invoke_after'
 | /home/ares/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/activesupport-5.1.4/lib/active_support/callbacks.rb:132:in `run_callbacks'
 | /home/ares/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/activesupport-5.1.4/lib/active_support/callbacks.rb:827:in `_run_save_callbacks'
 | /home/ares/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/activerecord-5.1.4/lib/active_record/callbacks.rb:336:in `create_or_update'
 | /home/ares/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/activerecord-5.1.4/lib/active_record/persistence.rb:129:in `save'
 | /home/ares/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/activerecord-5.1.4/lib/active_record/validations.rb:44:in `save'
 | /home/ares/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/activerecord-5.1.4/lib/active_record/attribute_methods/dirty.rb:35:in `save'
 | /home/ares/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/activerecord-5.1.4/lib/active_record/transactions.rb:308:in `block (2 levels) in save'
 | /home/ares/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/activerecord-5.1.4/lib/active_record/transactions.rb:384:in `block in with_transaction_returning_status'
 | /home/ares/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/activerecord-5.1.4/lib/active_record/connection_adapters/abstract/database_statements.rb:235:in `block in transaction'
 | /home/ares/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/activerecord-5.1.4/lib/active_record/connection_adapters/abstract/transaction.rb:194:in `block in within_new_transaction'
 | /home/ares/.rbenv/versions/2.4.0/lib/ruby/2.4.0/monitor.rb:214:in `mon_synchronize'
 | /home/ares/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/activerecord-5.1.4/lib/active_record/connection_adapters/abstract/transaction.rb:191:in `within_new_transaction'
 | /home/ares/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/activerecord-5.1.4/lib/active_record/connection_adapters/abstract/database_statements.rb:235:in `transaction'
 | /home/ares/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/activerecord-5.1.4/lib/active_record/transactions.rb:210:in `transaction'
 | /home/ares/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/activerecord-5.1.4/lib/active_record/transactions.rb:381:in `with_transaction_returning_status'
 | /home/ares/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/activerecord-5.1.4/lib/active_record/transactions.rb:308:in `block in save'
 | /home/ares/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/activerecord-5.1.4/lib/active_record/transactions.rb:323:in `rollback_active_record_state!'
 | /home/ares/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/activerecord-5.1.4/lib/active_record/transactions.rb:307:in `save'
 | /home/ares/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/activerecord-5.1.4/lib/active_record/suppressor.rb:42:in `save'
 | /home/ares/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/bullet-5.7.0/lib/bullet/active_record5.rb:5:in `save'
 | /home/ares/Projekty/Zdrojaky/foreman/app/controllers/api/v2/trends_controller.rb:21:in `create'
```

perhaps the fix should go to `foreman_trend.rb:17:in `create_values'`